vault/acme_billing_system_view.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+// See comment in command/format.go
+const hopeDelim = "♨"
+
+type acmeBillingSystemViewImpl struct {
+	extendedSystemView
+	logical.ManagedKeySystemView
+
+	core  *Core
+	entry *MountEntry
+}
+
+var _ logical.ACMEBillingSystemView = (*acmeBillingSystemViewImpl)(nil)
+
+func (c *Core) NewAcmeBillingSystemView(sysView interface{}, managed logical.ManagedKeySystemView) *acmeBillingSystemViewImpl {
+	es := sysView.(extendedSystemViewImpl)
+	des := es.dynamicSystemView
+
+	return &acmeBillingSystemViewImpl{
+		extendedSystemView:   es,
+		ManagedKeySystemView: managed,
+		core:                 c,
+		entry:                des.mountEntry,
+	}
+}
+
+func (a *acmeBillingSystemViewImpl) CreateActivityCountEventForIdentifiers(ctx context.Context, identifiers []string) error {
+	// Fake our clientID from the identifiers, but ensure it is
+	// independent of ordering.
+	//
+	// TODO: Because of prefixing currently handled by AddEventToFragment,
+	// we do not need to ensure it is globally unique.
+	sort.Strings(identifiers)
+	joinedIdentifiers := "[" + strings.Join(identifiers, "]"+hopeDelim+"[") + "]"
+	identifiersHash := sha256.Sum256([]byte(joinedIdentifiers))
+	clientID := base64.RawURLEncoding.EncodeToString(identifiersHash[:])
+
+	// Log so users can correlate ACME requests to client count tokens.
+	activityType := "acme"
```

Comment:
Would you mind making this a const? It will be useful for the `client_type` changes that I'm working on. No worries if not. I can make that change as well.

Reply:
Sure, I can make it a const, but I was a little unclear where I wanted it to live. I think ideally, _if_ we were to hoist it to a [proper interface](https://github.com/hashicorp/vault/pull/20520#discussion_r1185452458) it should probably live in `sdk/logical/` and each consumer of the interface would then define their own constant there that everyone could rely on?

But since this is an internal-ish interface only for use by ACME right now, I was thinking that leaving it here would be good for now...

Thoughts?

Reply:
Works for me!